core/src/test/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceServiceCreateSpec.scala

```diff
@@ -0,0 +1,849 @@
+      mock[BillingProfileManagerDAO],
+      samDAO,
+      MultiCloudWorkspaceConfig(MultiCloudWorkspaceManagerConfig("app", 60 seconds, 120 seconds), mock[AzureConfig]),
+      leonardoDAO,
+      "MultiCloudWorkspaceService-test",
+      mock[WorkspaceManagerResourceMonitorRecordDao],
+      workspaceRepository,
+      mock[BillingRepository]
+    )
+
+    val result: Workspace = Await.result(
+      service.createMultiCloudWorkspaceInt(workspaceRequest, workspaceId, billingProfile, testContext),
+      Duration.Inf
+    )
+
+    result.name shouldBe name
+    result.workspaceType shouldBe WorkspaceType.McWorkspace
+    result.namespace shouldEqual namespace
+    verify(workspaceManagerDAO)
+      .createWorkspaceWithSpendProfile(
+        ArgumentMatchers.eq(workspaceId),
+        ArgumentMatchers.eq(name),
+        ArgumentMatchers.eq(billingProfile.getId.toString),
+        ArgumentMatchers.eq(namespace),
+        any[Seq[String]],
+        ArgumentMatchers.eq(CloudPlatform.AZURE),
+        ArgumentMatchers.eq(MultiCloudWorkspaceService.buildPolicyInputs(workspaceRequest)),
+        ArgumentMatchers.eq(testContext)
+      )
+  }
+
+  behavior of "buildPolicyInputs"
+
+  it should "transform the policy inputs from the request" in {
+    val requestPolicies = List(
+      WorkspacePolicy("group-constraint", "terra", List(Map("group" -> "myFakeGroup"))),
+      WorkspacePolicy("region-constraint", "other-namespace", List(Map("key1" -> "value1"), Map("key2" -> "value2")))
+    )
+    val request = WorkspaceRequest(namespace, name, Map.empty, policies = Some(requestPolicies))
+
+    val policies = MultiCloudWorkspaceService.buildPolicyInputs(request)
+
+    policies shouldBe Some(
+      new WsmPolicyInputs()
+        .inputs(
+          Seq(
+            new WsmPolicyInput()
+              .name("group-constraint")
+              .namespace("terra")
+              .additionalData(List(new WsmPolicyPair().key("group").value("myFakeGroup")).asJava),
+            new WsmPolicyInput()
+              .name("region-constraint")
+              .namespace("other-namespace")
+              .additionalData(
+                List(new WsmPolicyPair().key("key1").value("value1"),
+                     new WsmPolicyPair().key("key2").value("value2")
+                ).asJava
+              )
+          ).asJava
+        )
+    )
+  }
+
+  it should "create a policy for the protected data flag" in {
+    val request = WorkspaceRequest(namespace, name, Map.empty, protectedData = Some(true))
+
+    val policies = MultiCloudWorkspaceService.buildPolicyInputs(request)
+
+    policies shouldBe Some(
+      new WsmPolicyInputs()
+        .inputs(
+          Seq(
+            new WsmPolicyInput()
+              .name("protected-data")
+              .namespace("terra")
+              .additionalData(List().asJava)
+          ).asJava
+        )
+    )
+  }
+
+  it should "merge the protected data policy with other inputs in the request" in {
```

Comment:
What happened to "fail with an improperly structured additional fields element on a policy and rollback workspace creation"?

https://github.com/broadinstitute/rawls/pull/2939/files#diff-b8196c4af84188faa6eb1f65595077eb0b25f6c06f5a8e64b5c60e4832b93118L660

Reply:
I thought there was a test in `WorkspaceModel`, but looks like I was mistaken. I'll add a test there.